### PR TITLE
AP_AHRS: make ExternalAHRS an AP_AHRS_Backend

### PR DESCRIPTION
**Pull request by peterbarker**
This starts to make all our existing backends inherit from AP_AHRS_Backend - rather than just AP_AHRS_DCM.

Future PRs make SITL and the EKFs backends as well.

 - [ ] We should probably test this on an actual ExternalAHRS, rather than trusting the simulated one.

### CODE REVIEW

**Comment by amilcarlucas** [comment]
```
Binary           text             data         bss           total
---------------  ---------------  -----------  ------------  ---------------
arduplane        1228 (+0.0706%)  0 (0.0000%)  4 (+0.0015%)  1232 (+0.0615%)
arducopter       728 (+0.0419%)   0 (0.0000%)  0 (0.0000%)   728 (+0.0364%)
ardurover        812 (+0.0520%)   0 (0.0000%)  4 (+0.0015%)  816 (+0.0447%)
antennatracker   764 (+0.0575%)   0 (0.0000%)  4 (+0.0015%)  768 (+0.0483%)
arducopter-heli  736 (+0.0429%)   0 (0.0000%)  0 (0.0000%)   736 (+0.0372%)
ardusub          792 (+0.0518%)   0 (0.0000%)  0 (0.0000%)   792 (+0.0442%)
blimp            676 (+0.0535%)   0 (0.0000%)  4 (+0.0015%)  680 (+0.0446%)
```

**Comment by khancyr** [comment]
yep, much better than bloatly !

File: libraries/AP_AHRS/AP_AHRS.cpp [hunk 1]
```diff
@@ -1238,9 +1231,11 @@ float AP_AHRS::groundspeed(void)
 #if HAL_NAVEKF3_AVAILABLE
     case EKFType::THREE:
 #endif
+        break;
```
Comment (tridge) [inline libraries/AP_AHRS/AP_AHRS.cpp:1234]: break inside #if ?

**Review by tridge — CHANGES_REQUESTED** [review]
need updated flash cost

File: libraries/AP_AHRS/AP_AHRS_External.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    posD = -(loc.alt - orgn.alt)*0.01;
+    return true;
+}
+
+bool AP_AHRS_External::get_velocity_NED(Vector3f &vec) const
+{
+    return AP::externalAHRS().get_velocity_NED(vec);
+}
+
+bool AP_AHRS_External::get_vert_pos_rate(float &velocity) const
```
Comment (tridge) [inline libraries/AP_AHRS/AP_AHRS_External.cpp:103]: change to pos_down_rate and fix sagetech user

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]
approved as long as sagetech and api fixed in followup PR

**Pull request merged**